pyrasterframes/src/main/python/tests/RasterSourceTest.py

```diff
@@ -211,3 +211,9 @@ def test_catalog_named_arg(self):
         df = self.spark.read.raster(catalog=self.path_pandas_df(), catalog_col_names=['b1', 'b2', 'b3'])
         self.assertEqual(len(df.columns), 7)  # three path cols, three tile cols, and geo
         self.assertTrue(df.select('b1_path').distinct().count() == 3)
+
+    def test_spatial_partitioning(self):
+        df = self.spark.read.raster(self.path(1, 1), spatial_index_partitions=True)
+        self.assertTrue('spatial_index' in df.columns)
+        # Other tests?
```

Comment:
maybe try passing in the following values with the expected results

int(0) -> no spatial index
float(99) -> df.rdd.getNumPartitions == 99
str("42.0") -> df.rdd.getNumPartitions == 42
False -> no spatial index